altos-rust/port/cortex-m0/src/peripheral/usart/control.rs

```diff
@@ -201,36 +442,69 @@ impl CR1 {
 /// CR2
 // ------------------------------------
 
-/// There are four stop bit settings: .5, 1, 1.5, 2
-#[derive(Copy, Clone)]
+/// Defines the possible StopLength configurations for the Usart.
+/* Half: 0.5 stop bit
+ * One: 1 stop bit
+ * OneAndHalf: 1.5 stop bits
+ * Two: 2 stop bits
+ */
+#[derive(Copy, Clone, Debug)]
 pub enum StopLength {
     Half,
     One,
     OneAndHalf,
     Two,
 }
 
-#[derive(Copy, Clone)]
+/// Defines the base address variable for control register two.
+#[derive(Copy, Clone, Debug)]
 struct CR2 {
     base_addr: *const u32,
 }
 
+/// Implements the Register trait for Usartx_CR2.
+/// Stores base address for control register two, which is the address
+/// of the Usart being used to access this register. Uses the base address
+/// combined with the register offset to calculate the register address.
 impl Register for CR2 {
+    /* Sets the base address as the Usart address.
+     * Returns itself to the calling routine.
+     */
     fn new(base_addr: *const u32) -> Self {
         CR2 { base_addr: base_addr }
     }
 
+    /* Helper function to calcluate the address of CR2.
+     * Supplies the base address to the `addr()` Register routine.
+     * Used in conjunction with the 'mem_offset' function below.
+     */
     fn base_addr(&self) -> *const u32 {
         self.base_addr
     }
 
+    /* Helper function to calculate the address of CR2.
+     * Supplies the CR2 offset address back to the 'addr()' Register routine.
+     * Used in conjuction with the 'base_addr' function above.
+     */
```

Comment:
Yeah... this is all just repeated information that doesn't add anything to a developer's understanding. Behavior of trait functions should be documented on the trait itself